include/tty/vga.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _VGA_H
+#define _VGA_H
+
+#include "kernel/kernel.h"
+#include "machine/pt.h"
+
+#define VGA_PHYS_ADDR 0xb8000
+#define VGA_VIRT_BASE phys_to_virt(VGA_PHYS_ADDR)
+
+#define VGA_HEIGHT 80
+#define VGA_WIDTH 20
+
+enum vga_color {
+
+	VGA_COLOR_BLACK 	= 0, 
+	VGA_COLOR_BLUE		= 1,
+	VGA_COLOR_GREEN		= 2,
+	VGA_COLOR_CYAN		= 3,
+	VGA_COLOR_RED		= 4,
+	VGA_COLOR_MAGENTA	= 5,
+	VGA_COLOR_BROWN		= 6,
+	VGA_COLOR_LIGHT_GREY	= 7,
+	VGA_COLOR_DARK_GREY	= 8,
+	VGA_COLOR_LIGHT_BLUE	= 9,
+	VGA_COLOR_LIGHT_GREEN	= 10,
+	VGA_COLOR_LIGHT_CYAN	= 11,
+	VGA_COLOR_LIGHT_RED	= 12,
+	VGA_COLOR_LIGHT_MAGENTA = 13,
+	VGA_COLOR_LIGHT_BROWN	= 14,
+	VGA_COLOR_WHITE		= 15,
+
+};
+
+static inline u8 vga_entry_color(enum vga_color fg, enum vga_color bg)
+{
+	return fg | bg << 4;
```

Comment:
i am confused?

is it:
```suggestion
	return (fg | bg) << 4;
```

or

```suggestion
	return fg | (bg << 4);
```

shall we use parentheses

Reply:
from my understanding in c `|` has a lower precedence than `<<`, so i suppose that he meant `return fg | (bg << 4);` which is basically the same as return `fg | bg << 4;`